src/overlays/actors/ovl_Mir_Ray/z_mir_ray.c

```diff
@@ -30,25 +42,537 @@ const ActorInit Mir_Ray_InitVars = {
+    this->colliderSph.list->dim.worldSphere.center.x = temp_vec.x + this->sourcePt.x;
+    this->colliderSph.list->dim.worldSphere.center.y = temp_vec.y + this->sourcePt.y;
+    this->colliderSph.list->dim.worldSphere.center.z = temp_vec.z + this->sourcePt.z;
+    this->colliderSph.list->dim.worldSphere.radius = temp_DataEntry->unk_14 * this->colliderSph.list->dim.scale;
+}
+
+// Set up a light point between source point and reflection point. Reflection point is the pool point (for windows) or
+// at the player position (for mirrors) void MirRay_SetupLights(MirRay* this, GlobalContext* globalCtx) {
+void MirRay_MakeShieldLight(MirRay* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    MirRayDataEntry* temp_DataEntry = &sMirRayData[this->actor.params];
+    Vec3f reflectionPt;
+    Vec3s lightPt;
+
+    if (MirRay_CheckInFrustum(&this->sourcePt, &this->poolPt, player->actor.posRot.pos.x,
+                              player->actor.posRot.pos.y + 30.0f, player->actor.posRot.pos.z, this->sourceEndRad,
+                              this->poolEndRad)) {
+
+        if (temp_DataEntry->params & 8) { // Light beams from mirrors
+            Math_Vec3f_Diff(&player->actor.posRot.pos, &this->sourcePt, &reflectionPt);
+        } else { // Light beams from windows
+            Math_Vec3f_Diff(&this->poolPt, &this->sourcePt, &reflectionPt);
+        }
+
+        lightPt.x = (temp_DataEntry->unk_18 * reflectionPt.x) + this->sourcePt.x;
+        lightPt.y = (temp_DataEntry->unk_18 * reflectionPt.y) + this->sourcePt.y;
+        lightPt.z = (temp_DataEntry->unk_18 * reflectionPt.z) + this->sourcePt.z;
+
+        // Fade up
+        Math_ApproxS(&this->lightPointRad, temp_DataEntry->lgtPtMaxRad, 6);
+        Lights_PointNoGlowSetInfo(&this->lightInfo, lightPt.x, lightPt.y, lightPt.z, temp_DataEntry->r,
+                                  temp_DataEntry->g, temp_DataEntry->b, this->lightPointRad);
+    } else {
+        // Fade down
+        Math_ApproxS(&this->lightPointRad, 0, 6);
+        Lights_PointSetColorAndRadius(&this->lightInfo, temp_DataEntry->r, temp_DataEntry->g, temp_DataEntry->b,
+                                      this->lightPointRad);
+    }
+}
+
+void MirRay_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    MirRay* this = THIS;
+    MirRayDataEntry* temp_DataEntry = &sMirRayData[this->actor.params];
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    // Generation of reflectable light!
+    osSyncPrintf("反射用 光の発生!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
+    LogUtils_LogThreadId("../z_mir_ray.c", 0x206);
+    osSyncPrintf("this->actor.arg_data = %d\n", this->actor.params);
+
+    if (this->actor.params >= 0xA) {
+        LogUtils_LogThreadId("../z_mir_ray.c", 0x209);
+        // Reflected light generation failure
+        osSyncPrintf("\"反射光 発生失敗\" = %s\n", "反射光 発生失敗");
```

Comment:
And we also have a variant for strings
```suggestion
        // Reflected light generation failure
        LOG_STRING("反射光 発生失敗", "../z_mir_ray.c", 521);
```